connectors/sources/sharepoint_online.py

```diff
@@ -1005,6 +1019,15 @@ def get_default_configuration(cls):
                 "type": "bool",
                 "value": False,
             },
+            "fetch_unique_page_permissions": {
+                "depends_on": [{"field": "use_document_level_security", "value": True}],
+                "display": "toggle",
+                "label": "Fetch unique page permissions",
+                "order": 10,
+                "tooltip": "Enable this option to fetch unique page permissions. This setting can increase sync time. If this setting is disabled a page will inherit permissions from its parent site.",
+                "type": "bool",
+                "value": True,
+            },
```

Comment:
I've just recently merged functional tests for SPO - can you add this field into the `connector.json` for it?

Reply:
It's already added 👍

Reply:
Additionally (I commented in another issue), you might wanna decrement order by 1 cause of https://github.com/elastic/connectors-python/pull/1255#pullrequestreview-1528292491

Reply:
Good point, will adjust it in this PR and in https://github.com/elastic/connectors-python/pull/1259